src/analysis/processing/qgsalgorithmrasterlogicalop.cpp

```diff
@@ -0,0 +1,309 @@
+/***************************************************************************
+                         qgsalgorithmrasterlogicalop.cpp
+                         ---------------------
+    begin                : March 2019
+    copyright            : (C) 2019 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmrasterlogicalop.h"
+#include "qgsrasterprojector.h"
+#include "qgsrasterfilewriter.h"
+#include "qgsrasteranalysisutils.h"
+#include <algorithm>
+
+///@cond PRIVATE
+
+
+QStringList QgsRasterBooleanLogicAlgorithmBase::tags() const
+{
+  return QObject::tr( "logical,boolean" ).split( ',' );
+}
+
+QString QgsRasterBooleanLogicAlgorithmBase::group() const
+{
+  return QObject::tr( "Raster analysis" );
+}
+
+QString QgsRasterBooleanLogicAlgorithmBase::groupId() const
+{
+  return QStringLiteral( "rasteranalysis" );
+}
+
+void QgsRasterBooleanLogicAlgorithmBase::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterMultipleLayers( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layers" ), QgsProcessing::TypeRaster ) );
+
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "REF_LAYER" ), QObject::tr( "Reference layer" ) ) );
```

Comment:
On top of the reference layer, I'm wondering whether an optional extent parameter to restrict the output to a given extent would be useful here.

Reply:
I can see how this would be useful for some algorithms - but for these particular two I don't see a pressing need for this (as opposed to say - hillshade/aspect/etc, where they are more likely to be run on a subset of a larger/global layer). And I'd rather defer this until there's a particular need as I think it's something we should consider a unified approach for all algorithms to (e.g. maybe an arc style "processing" environment option).